src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerTest.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.companieshouse.itemhandler.kafka;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.kafka.consumer.resilience.CHKafkaResilientConsumerGroup;
+import uk.gov.companieshouse.kafka.exceptions.SerializationException;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+public class OrdersKafkaConsumerTest {
+    @InjectMocks
+    private OrdersKafkaConsumer ordersKafkaConsumer;
+    @Mock
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupMain;
+    @Mock
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupRetry;
+
+    @Test(expected = Exception.class)
+    public void mainListenerExceptionIsCorrectlyHandled() throws InterruptedException, ExecutionException, SerializationException {
+        doThrow(Exception.class).when(ordersKafkaConsumer).processOrderReceived(anyString());
+        verify(chKafkaConsumerGroupMain, times(3)).retry(anyInt(), any());
+    }
+
+    @Test(expected = Exception.class)
+    public void retryListenerExceptionIsCorrectlyHandled() throws InterruptedException, ExecutionException, SerializationException {
+        doThrow(Exception.class).when(ordersKafkaConsumer).processOrderReceivedRetry(anyString());
+        verify(chKafkaConsumerGroupRetry, times(3)).retry(anyInt(), any());
+    }
+}
```

Comment:
* These two tests appear to have `Given` and `Then` elements, but no `When` element? 
* Perhaps using `assertThrows` would make this less likely than when using `@Test(expected = Exception.class)`?
* Also, looking at this now, I think one problem with throwing and catching/expecting an `Exception(.class)` even in a test is that we cannot be sure that the exception is the one we intended/expected?

Reply:
This has been updated.